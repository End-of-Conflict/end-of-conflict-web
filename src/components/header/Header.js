/**
 * Header
 * @flow
 */
import React from 'react';
import { useRouteMatch } from "react-router-dom";

type Props = {
  title: string,
};

const Header = (props: Props): React$Element<"header"> => {
  let match = useRouteMatch("/chapters/en/:chapter");
  const { title } = props;
  const heading = !match || match.url === '/' ? title : `EoC: ${title}`;
  document.title =  !match || match.url === '/' ? title : `End of Conflict | Chapter ${match.params.chapter}: ${title}`;

  return (
    <header>
      <h1>{heading}</h1>
    </header>
  );
}

Header.defaultProps = {
  title: 'The end of conflict',
};

export default Header;
