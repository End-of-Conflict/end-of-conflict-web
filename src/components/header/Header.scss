@import '../../base/breakpoints';
@import '../../base/colors';

.header {
  background: $alto;
  position: relative;

  header {
    margin: 0 auto;
    padding: 1rem;
  
    @include breakpoint(sm) {
      width: 42rem;
      display: flex;
      align-items: center;
      justify-content: space-between;
      padding: 1rem 0;
    }
  
    .book-title {
      margin: 0 0 0 1.5rem;
      font-weight: 700;
      text-decoration: none;
      color: $mine-shaft;

      @include breakpoint(sm) {
        margin: 0;
      }
  
      &:visited {
        color: $mine-shaft;
      }
    }
  }
}

