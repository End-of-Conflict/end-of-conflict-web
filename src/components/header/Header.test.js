import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from 'enzyme-to-json';
import Header from './Header';
expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useRouteMatch: () => ({
    url: '/en/0',
    params: { chapter: 0 }
  }),
}));
configure({ adapter: new Adapter() });

describe('Renders', () => {
  test('Header Introduction', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
