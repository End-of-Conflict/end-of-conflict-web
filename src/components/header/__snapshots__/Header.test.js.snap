// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Header Introduction 1`] = `
<header>
  <h1>
    EoC: The end of conflict
  </h1>
</header>
`;
